main.py

```diff
@@ -52,35 +53,42 @@ def echo(message, history, state):
     chat_history = chat_history + user_message
     state['messages'] = chat_history
 
-    response = client.models.generate_content(
-        model="gemini-1.5-flash",
-        contents=state['messages'],
-    )
-    model_response = response.text
-
+    response_chunks = ""
+    async for chunk in await client.aio.models.generate_content_stream(
```

Comment:
Prefer using a variable for `await client.aio.models.generate_content_stream()`.